### PR TITLE
Transfer print money between accounts (v1)

### PR DESCRIPTION
**Pull request by katkodominik**
Now users can send money from their print accounts to someone else's. (issue #94)
Future updates:

- I think the sent money should be appear as free pages, but idk how much does one free page worth.

- A notification email would be nice.

### CODE REVIEW

File: app/Http/Controllers/PrintController.php [hunk 1]
```diff
@@ -52,6 +52,26 @@ public function print(Request $request) {
             return back()->withErrors(['print' => __('print.error_printing')]);
         }
     }
+    public function sendPages(Request $request) {
```
Comment (steakhal) [inline app/Http/Controllers/PrintController.php]: transferBalance?

File: app/Http/Controllers/PrintController.php [hunk 2]
```diff
@@ -52,6 +52,26 @@ public function print(Request $request) {
...
+        $validator = Validator::make($request->all(), [
+            'balance' => 'required|integer',
```
Comment (steakhal) [inline app/Http/Controllers/PrintController.php]: I don't think a negative balance transfer should be allowed.
Validate the proper value range.

File: routes/web.php [hunk 3]
```diff
@@ -36,6 +36,7 @@
 
     Route::get('/print', 'PrintController@index')->name('print');
     Route::post('/print/modify_balance', 'PrintController@modifyBalance')->name('print.modify')->middleware('can:print.modify');
+    Route::post('/print/send_pages', 'PrintController@sendPages')->name('print.send');
```
Comment (steakhal) [inline routes/web.php]: There is confusion all over the place.
If we transfer balance (money), why we call this feature _sending pages_...

If we are sending money, we should be clear about that and not put it under the _print_ section.
Maybe a link with a description and a logo would do the job redirecting the user to _transferring balance_ page.

File: resources/views/print/send.blade.php [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<div class="card">
+    <div class="card-header">@lang('print.transfer_money')</div>
+    <div class="card-body">
+        <form method="POST" action="{{ route('print.send') }}">
```
Comment (steakhal) [inline resources/views/print/send.blade.php]: **send** is not descriptive enough, just look 2 lines above - that is descriptive.

File: resources/views/print/send.blade.php [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+            @csrf
+            <select class="selectpicker" data-live-search="true" id="receiver_id" name="receiver_id" title="@lang('print.user')">
+                @foreach ($users as $user)
+                <option value="{{ $user->id }}">{{ $user->name }}</option>
+                @endforeach
+            </select>
+            <input id="balance" name="balance" type="number">
```
Comment (luksan47) [inline resources/views/print/send.blade.php]: You could add a placeholder here (see #155), so the user knows what to write here. I think it looks even better on mobile.

File: resources/views/print/send.blade.php [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<div class="card">
+    <div class="card-header">@lang('print.transfer_money')</div>
+    <div class="card-body">
+        <form method="POST" action="{{ route('print.send') }}">
+            @csrf
+            <select class="selectpicker" data-live-search="true" id="receiver_id" name="receiver_id" title="@lang('print.user')">
```
Comment (luksan47) [inline resources/views/print/send.blade.php]: Just a note: we have a search blade that could be included here after #152 is resolved.

**Comment by luksan47** [comment]
Thanks @katkodominik for adding this!

**Pull request merged**